ground/src/main/java/com/google/android/ground/persistence/remote/firebase/FirestoreDataStore.kt

```diff
@@ -77,7 +78,11 @@ internal constructor(
 
   /** Calls Cloud Function to refresh the current user's profile info in the remote database. */
   override suspend fun refreshUserProfile() {
-    firebaseFunctions.getHttpsCallable(PROFILE_REFRESH_CLOUD_FUNCTION_NAME).call().await()
+    try {
+      firebaseFunctions.getHttpsCallable(PROFILE_REFRESH_CLOUD_FUNCTION_NAME).call().await()
+    } catch (e: FirebaseFunctionsException) {
+      Timber.w(e, "Failed to refresh user data")
```

Comment:
This will be a valid error when the user is actually online. Instead of eating the exception, can we handle if downstream in the caller? 

A broader problem is that we shouldn't be trying to load ToS and updating user profile when offline; waiting for timeout takes time, which makes the app appear to load slowly. Can we skip those steps when offline and the user is already signed in?

Reply:
Makes sense! Thanks for the suggestion. I'll update this PR.

Reply:
Done